plugins/optimization-detective/storage/class-od-url-metrics-post-type.php

```diff
@@ -298,15 +298,24 @@ public static function schedule_garbage_collection(): void {
 	 * @since 0.1.0
 	 */
 	public static function delete_stale_posts(): void {
-		$one_month_ago = gmdate( 'Y-m-d H:i:s', strtotime( '-1 month' ) );
+		/**
+		 * Filters the time for the garbage collection ttl for the stale posts.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @return int Number of previous months to check the older post for url metrics.
+		 */
+		$updated_time = (int) apply_filters( 'od_url_metric_garbage_collection_ttl', 1 );
+		$time_string  = sprintf( '-%d month', absint( $updated_time ) );
+		$before_time  = gmdate( 'Y-m-d H:i:s', (int) strtotime( $time_string ) );
```

Comment:
```suggestion
		/**
		 * Filters the expiration time (TTL) after which an unmodified od_url_metrics post not touched since that time will be garbage collected.
		 *
		 * @since n.e.x.t
		 *
		 * @return int TTL for garbage collection in seconds. Defaults to 3 months.
		 */
		$ttl = (int) apply_filters( 'od_url_metric_garbage_collection_ttl', 3 * MONTH_IN_SECONDS );
		if ( $ttl <= 0 ) {
			return;
		}
		$before_time  = gmdate( 'Y-m-d H:i:s', time() - $ttl );
```

Reply:
This can add a link to the `hooks.md` docs, which also mention:

> A filtered value of zero (0) will prevent garbage collection from occurring.

Also include the sample code mentioned above.

See https://github.com/WordPress/performance/pull/1940 for how I think we can leverage `@link` phpdoc tags in the the filter docs to direct developers to the docs on GitHub.